wordpress-mobile/gutenberg-mobile
# Add ability to toggle html/visual modes through the bridge

This change adds switch to html/visual mode messaging to the bridge and demonstrates it on the example app. 
On RN side: Updates the BlockManager to handle TOGGLE_BLOCK_MODE updates and re-render its contents.

<img width="174" alt="screen shot 2018-11-26 at 16 33 52" src="https://user-images.githubusercontent.com/5032900/49017222-60bba280-f199-11e8-8317-607f92fe4a94.png">

Part of #182

**To Test**

Test 1:

- Run yarn install & yarn start and yarn ios as usual to open the example app
- Tap (...) button on the rightside of the navigation bar
- Tap "Switch to HTML"
- Verify that HTML editor is shown
- Tap (...) button on the rightsize of the navigation bar again
- Tap "Switch to Visual"
- Verify that visual editor is shown

Test 2:

- Reopen the example app
- Tap (...) button on the rightside of the navigation bar
- Tap "Switch to HTML"
- Edit something in html edit mode(do not switch back to the visual mode)
- Tap Save in the navigation bar
- Verify that you see the edited version of the html in Xcode logs

Test 3:

- Reopen the example app
- Tap (...) button on the rightside of the navigation bar
- Tap "Switch to HTML"
- Edit something in html edit mode
- Tap (...) button on the rightsize of the navigation bar again
- Tap "Switch to Visual"
- Verify that you see the edited version of the html in visual mode
- Tap Save in the navigation bar
- Verify that you see the edited version of the html in Xcode logs

Also:

- Verify that both "Inspect blocks" and "Switch to html mode" switches are removed.
